### ISSUE TITLE
Add a `// forge-import: Contract.sol` comment directive

### ISSUE THREAD

**Issue by adhusson** [foundry-rs/foundry#7025]
### Component

Forge

### tldr
The effect of `// forge-import: Contract.sol` is to add a contract to the list of contracts to be compiled (but it does not import it in the file containing the directive.

### Rationale

`vm.getCode("Contract.sol")` can be used to import the bytecode of a contract that was compiled using a [different compiler version](https://github.com/foundry-rs/foundry/pull/4) (to match an onchain deployment for instance). If the contract is not used anywhere else, an additional [mock file](https://github.com/foundry-rs/foundry/issues/1163#issuecomment-1102741463) can be used to make sure the contract is compiled by `forge build`.

This stops working when the foundry project that uses `getCode` is itself used as a dependency. The mock file is usually not imported, so `Contract.sol` is not compiled at all.

The `forge-import` directive would statically add a contract to the compilation plan.

Related: #6099

**Comment by sambacha** [comment]
Please no, this feels like more like a foundry.toml thingy than a comment directive. 

People will use this for more than this stated use case, and forge has yet to address its dependency management technical debt.

**Comment by adhusson** [comment]
What other use cases do you have in mind? Definitely possible to add a foundry.toml option instead, but having the import right next to the `getCode` call looks easier to maintain.

**Comment by zerosnacks** [comment]
Not a fan of the proposed comment directive approach but I can see how this feature could be useful, echoing https://github.com/foundry-rs/foundry/pull/7738

Perhaps a mapping similar to how libraries can be linked in `foundry.toml`?:

```toml
imports = [
  "path/to/file.sol:ContractName"
]
```

and then you can reference the artifact by name

cc @klkvr / @mattsse

**Comment by grandizzy** [comment]
@adhusson would the proposed PR https://github.com/foundry-rs/foundry/pull/8668 solve the scenario you're looking for?

**Comment by adhusson** [comment]
AFAICT no as it does not force the compilation of additional files (instead it discriminates files already scheduled for compilation).

Combined with @zerosnacks's [suggestion](https://github.com/foundry-rs/foundry/issues/7025#issuecomment-2221072002) yes it would help.
